alfresco-search/src/test/java/org/alfresco/solr/tracker/DistributedAlfrescoSolrTrackerRaceTest.java

```diff
@@ -46,77 +46,77 @@
+//        txn.setCommitTimeMs(backdatedCommitTimeMs);
+//
+//        //Next create two nodes to update for the transaction
+//        Node folderNode = getNode(txn, acl, Node.SolrApiNodeStatus.UPDATED);
+//        Node fileNode = getNode(txn, acl, Node.SolrApiNodeStatus.UPDATED);
+//        Node errorNode = getNode(txn, acl, Node.SolrApiNodeStatus.UPDATED);
+//
+//
+//        //Next create the NodeMetaData for each node. TODO: Add more metadata
+//        NodeMetaData folderMetaData = getNodeMetaData(folderNode, txn, acl, "mike", null, false);
+//        NodeMetaData fileMetaData   = getNodeMetaData(fileNode, txn, acl, "mike", ancestors(folderMetaData.getNodeRef()), false);
+//        //The errorNodeMetaData will cause an exception.
+//        NodeMetaData errorMetaData   = getNodeMetaData(errorNode, txn, acl, "lisa", ancestors(folderMetaData.getNodeRef()), true);
+//
+//        //Index the transaction, nodes, and nodeMetaDatas.
+//        //Note that the content is automatically created by the test framework.
+//        indexTransaction(txn,
+//            list(errorNode, folderNode, fileNode),
+//            list(errorMetaData, folderMetaData, fileMetaData));
+//
+//
+//        indexAclChangeSet(aclChangeSet,
+//            list(acl, acl2),
+//            list(aclReaders, aclReaders2));
+//
+//        System.out.println(backdatedCommitTimeMs + ":" + txnCommitTimeMs);
+//
+//        BooleanQuery.Builder builder = new BooleanQuery.Builder();
+//        builder.add(new BooleanClause(new TermQuery(new Term(QueryConstants.FIELD_SOLR4_ID, "TRACKER!STATE!ACLTX")), BooleanClause.Occur.MUST));
+//        builder.add(new BooleanClause(LegacyNumericRangeQuery.newLongRange(QueryConstants.FIELD_S_ACLTXID, aclChangeSet.getId(), aclChangeSet.getId() + 1, true, false), BooleanClause.Occur.MUST));
+//        BooleanQuery waitForQuery = builder.build();
+//        waitForDocCountAllCores(waitForQuery, 1, 80000);
+//
+//        /*
+//        * Query the index for the content
+//        */
+//        //This acl should have one record in each core with DBID sharding
+//        waitForDocCountAllCores(new TermQuery(new Term(QueryConstants.FIELD_READER, "jim")), 1, 80000);
+//        waitForDocCount(new TermQuery(new Term("content@s___t@{http://www.alfresco.org/model/content/1.0}content", "world")), 2, 100000);
+//        waitForDocCount(new TermQuery(new Term("content@s___t@{http://www.alfresco.org/model/content/1.0}content", Long.toString(fileNode.getId()))), 1, 80000);
+//
+//
+//        //This will run the query on the control client and the cluster and compare the result.
+//        query(getDefaultTestClient(), true, "{\"locales\":[\"en\"], \"templates\": [{\"name\":\"t1\", \"template\":\"%cm:content\"}]}",
+//            params("q", "t1:world", "qt", "/afts", "shards.qt", "/afts", "start", "0", "rows", "6", "sort", "id asc"));
+//
+//    }
+//      TODO Investigate how this ever got the correct txn times
```

Comment:
Should a Jira issue be open for this investigation?
It's in doubt the correctness of such test?
 I normally discourage to merge commented code on master, can't we Ignore the test and add the related JIRA as reference?

Reply:
I remember an old version of JUnit which used to throw an exception is a test class had no test method. This is not the case (just tried :) ) but I usually follow the same approach mentioned by Alessandro.

Reply:
I will raise a ticket for DistributedAlfrescoSolrTrackerRaceTest